src/lib/storage/lz4_segment.cpp

```diff
@@ -0,0 +1,147 @@
+#include "lz4_segment.hpp"
+
+#include "resolve_type.hpp"
+#include "storage/vector_compression/base_compressed_vector.hpp"
+#include "utils/assert.hpp"
+#include "utils/performance_warning.hpp"
+
+#include "lib/lz4.h"
+
+namespace opossum {
+
+template <typename T>
+LZ4Segment<T>::LZ4Segment(const std::shared_ptr<const pmr_vector<char>>& compressed_data,
+                          const std::shared_ptr<const pmr_vector<bool>>& null_values,
+                          const std::shared_ptr<const pmr_vector<size_t>>& offsets, const int compressed_size,
```

Comment:
Why is `compressed_size` an int? I see why the return values in the code have to be signed, but here?